From bf857959df3c07b60d6952210b16b5ddbcbfb31e Mon Sep 17 00:00:00 2001
From: Xinyu Chen <b03824@freescale.com>
Date: Fri, 10 Feb 2012 15:55:59 +0800
Subject: ENGR00174212 isl29023: fix build error of isl29023 light sensor

the platform data and irq set type interface is changed.

Signed-off-by: Xinyu Chen <xinyu.chen@freescale.com>
---
 drivers/input/misc/isl29023.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/input/misc/isl29023.c b/drivers/input/misc/isl29023.c
index 8ea0deb..8a0c16b 100755
--- a/drivers/input/misc/isl29023.c
+++ b/drivers/input/misc/isl29023.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2011 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright (C) 2011-2012 Freescale Semiconductor, Inc. All Rights Reserved.
  */
 
 /*
@@ -866,7 +866,7 @@ static int __devinit isl29023_probe(struct i2c_client *client,
 	if (!data)
 		return -ENOMEM;
 
-	ls_data = (struct mxc_lightsensor_platform_data *)
+	ls_data = (struct fsl_mxc_lightsensor_platform_data *)
 	    (client->dev).platform_data;
 
 	data->client = client;
@@ -906,7 +906,7 @@ static int __devinit isl29023_probe(struct i2c_client *client,
 		goto exit_free_input;
 
 	/* set irq type to edge falling */
-	set_irq_type(client->irq, IRQF_TRIGGER_FALLING);
+	irq_set_irq_type(client->irq, IRQF_TRIGGER_FALLING);
 	err = request_irq(client->irq, isl29023_irq_handler, 0,
 			  client->dev.driver->name, data);
 	if (err < 0) {
-- 
1.7.9.5

